src/Book.ts

```diff
@@ -26,6 +33,20 @@ const GROUP = {
   },
 };
 
+function fromDistinctToAccounts(distinctResult: string[]) {
+  const accountsSet: Set<string> = new Set();
+  for (const result of distinctResult) {
+    const prev: string[] = [];
+    const paths: string[] = result.split(":");
+    for (const acct of paths) {
+      prev.push(acct);
+      accountsSet.add(prev.join(":"));
+    }
+  }
+
+  return Array.from(accountsSet);
+}
+
```

Comment:
fromDistinctToAccounts seems a little bit computational heavy.
Would like to see a benchmark for this function.

can we reduce the pressure, by reusing prev bytaking it out of the for loop and doing a prev.length = 0?

Also the join call is probably also expensive. 

It makes more sense to it like this:
```suggestion
function fromDistinctToAccounts(distinctResult: string[]) {
  const accountsSet: Set<string> = new Set();
  let i = 0;
  for (const result of distinctResult) {
    const paths: string[] = result.split(":");
    for (i = 0; i < paths.length; ++i) {
      switch (i) {
        case 0:
          accountsSet.add(paths[0]);
          break;
        case 1:
          accountsSet.add(`${paths[0]}:${paths[1]});
          break;
        case 2:
          accountsSet.add(`${paths[0]}:${paths[1]:${paths[2]});
          break;
        case 3:
          accountsSet.add(`${paths[0]}:${paths[1]:${paths[2]:${paths[3]});
          break;
       default:
         // is it till i or does it need to be i -1 or i+1?
         accountsSet.add(paths.slice(0, i).join(':'));
    }
  }

  return Array.from(accountsSet);
}
```

untested. But it should reduce some pressure because we avoid alot of intermediary arrays.

Reply:
Suggestion listened. However, I'd implemented even simpler and faster solution:

```js
function fromDistinctToAccounts(distinctResult: string[]) {
  const accountsSet: Set<string> = new Set();
  for (const fullAccountName of distinctResult) {
    const paths: string[] = fullAccountName.split(":");
    let path = paths[0];
    accountsSet.add(path);
    for (let i = 1; i < paths.length; ++i) {
      path += ":" + paths[i];
      accountsSet.add(path);
    }
  }
  return Array.from(accountsSet);
}
```

Reply:
Also, this function will be called very occasionally. There is no point optimising it.